File: src/main/java/coffee/amo/interactionkey/util/ItemUtil.java
package coffee.amo.interactionkey.util;

import net.minecraft.world.entity.item.ItemEntity;
import net.minecraft.world.entity.player.Player;

import java.util.Objects;

public class ItemUtil {
    public static ItemEntity getClosestItem(Player player) {
        return player.level.getEntities(null, player.getBoundingBox().inflate(2)).stream().map(s -> s instanceof ItemEntity? (ItemEntity) s : null).filter(Objects::nonNull).min((s1, s2) -> {
            double d1 = player.distanceToSqr(s1);
            double d2 = player.distanceToSqr(s2);
            return Double.compare(d1, d2);
        }).orElse(null);
    }
}

File: src/main/kotlin/coffee/amo/interactionkey/util/ItemUtil.kt
package coffee.amo.interactionkey.util

import net.minecraft.world.entity.Entity
import net.minecraft.world.entity.item.ItemEntity
import net.minecraft.world.entity.player.Player
import java.util.*

object ItemUtil {
    @JvmStatic
    fun getClosestItem(player: Player): ItemEntity? {
        return player.level.getEntities(null, player.boundingBox.inflate(2.0)).stream().map { s: Entity? -> if (s is ItemEntity) s else null }.filter { obj: ItemEntity? -> Objects.nonNull(obj) }.min { s1: ItemEntity?, s2: ItemEntity? ->
            val d1 = s1?.let { player.distanceToSqr(it) }
            val d2 = s2?.let { player.distanceToSqr(it) }
            d1!!.compareTo(d2!!);
        }.orElse(null)
    }
}